serpent-os/recipes
# fish: bash as login shell, fish as interactive shell

This is a companion update to the #482 and #508 PRs (to be landed after both) , which piggybacks onto the configuration changes and offers a single file that can optionally start fish as an interactive shell after sourcing the default stateless shell settings via bash.

This file can either be symlinked into into ~/.bashrc.d from /usr/share/defaults/profile.d or, alternatively, be sourced from the user's custom ~/.bashrc file.

**Context for the proposed solution**
- https://wiki.gentoo.org/wiki/Fish#Caveats
- https://wiki.archlinux.org/title/Fish#Modify_.bashrc_to_drop_into_fish 
- https://nixos.wiki/wiki/Fish

**Test Plan**

- Install fish
- ln -sv /usr/share/defaults/profile.d/99-exec-fish-as-interactive-shell.conf ~/.bashrc.d/99-exec-fish-as-interactive-shell.conf.sh
- Open a new virtual terminal and see fish getting started as the interactive shell w/starship prompt enabled ootb if it is installed
- Execute `bash` and see bash get started with the correct environment

What this looks like (I am using a slightly customised fish prompt):

```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
󰌽 ermo@snekws:~/.bashrc.d                                                                                                16:06
❯ 
```
